opps/core/models.py

```diff
@@ -236,11 +236,11 @@ def save(self, *args, **kwargs):
             if self.pk is not None:
                 old_object = model.objects.get(pk=self.pk)
                 if old_object.slug != self.slug:
-                    redirect = Redirect(
+                    redirect, created = Redirect.objects.get_or_create(
                         site=self.site,
-                        old_path=old_object.get_absolute_url(),
-                        new_path=self.get_absolute_url()
+                        old_path=old_object.get_absolute_url()
                     )
+                    redirect.new_path = self.get_absolute_url()
```

Comment:
Use the `defaults` instead of save the object again.
https://docs.djangoproject.com/en/1.9/ref/models/querysets/#get-or-create